nodestream_plugin_k8s/commands/list_command.py

```diff
@@ -33,13 +39,25 @@ def display_as_json(self, items_to_display: Iterable[PipelineDesiredState]):
         )
         self.line(json_string)
 
+    def add_development_args(self, items_to_display):
+        for item in items_to_display:
+            item.perpetual_concurrency = (
+                item.perpetual_concurrency if item.debug_enabled else 0
+            )
+            item.suspend = False if item.debug_enabled else True
+            yield item
+
     async def handle_async(self):
         project_manager = ProjectResourceManager(self.get_project())
         scope_name = self.option(SCOPE_NAME_OPTION.name)
         items_to_display = (
             mgr.desired_state
             for mgr in project_manager.get_managed_pipelines(scope_name)
         )
+        development_environment = self.option(DEVELOPMENT_ENVIRONMENT_OPTION.name)
+        if development_environment:
+            items_to_display = self.add_development_args(items_to_display)
+
```

Comment:
Generally boolean options don't leave you room to change things without a breaking release. What if we changed this PR a bit: 

1.) What if you could pass the environment name in from the CLI.
2.) That maps to a config file that says whether or not the environment is development.

The reason I suggest this is that in the future, this plugin is intended to take on the work of provisioning and managing cron jobs and deployments instead of leaving that to the user. This would give us a place to hang additional configuration such as what cluster, namespace, etc to manage when passing `--env dev` or `--env prod` for example.

Thoughts?

Reply:
So what we want is to have variables output from the nodestream k8s list command to have values from the annotations that we specify. For instance, in prd we want the value from the annotation called: nodestream_plugin_k8s_prd_conccurency to specify the concurrency. But in dev, we want the annotation:
nodestream_plugin_k8s_dev_concurrency to specify the concurrency. 

Additionally for the cron pipeliens:
nodestream_plugin_k8s_prd_suspend: False
nodestream_plugin_k8s_dev_suspend: True

So would this be additional data to include in the final json result of the command?
nodestream k8s list -e(--env) dev


k8s/dev.yaml: 
{
    namespace: "dev-namespace,
    cluster: "main-cluster-usw2"
    suspend: !from_annotation "dev_suspend"
    concurrency: !from_annotation "dev_concurrency
}


k8s/prd.yaml: 
{
    namespace: "prd-namespace,
    cluster: "main-cluster-usw2"
    suspend: !from_annotation "prd_suspend"
    concurrency: !from_annotation "prd_concurrency
    additional_thing: !from_annotation "stuff"
    other_thing: "stuff2"
}

Reply:
@angelosantos4 and I connected via Zoom. Here are our conclusions:

1. Do not require setting an environment file up. Fallback to the behavior as it is today. 
2. Use a configuration file like the one below. Lets not introduce tags yet, but rather specify the annotations to look for. 

```yaml
namespace: "dev-namespace"
cluster: "main-cluster-usw2"
suspend_annotation_name: nodestream_plugin_k8s_dev_suspend
concurrency_annotation_name: nodestream_plugin_k8s_dev_concurrency
```

Reply:
Here is my suggestion for implementation: 

1. Consider creating a `EnvironmentConfiguration` class to encapsulate the data/configuration above. 
2. Augment `ProjectResourceManager` to take in a `EnvironmentConfiguration` instance. 
3. Move `is_managable` logic from `PipelineResourceManager` to `EnvironmentConfiguration`. 
...probably.